feat: rename collector pods and eraser container

ashnamehrotra opened the pull request:
**What this PR does / why we need it**:
- Renames "collector" pods to "eraser" pods and changes "eraser" container name to "remover". If scheduled by ImageCollector, the pods will have label "collector", and if scheduled by ImageList they will have label "remover".
- Fixes helm upgrade tests.
- Fixes configmap update reconfiguration. 

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #453

**Special notes for your reviewer**:

ashnamehrotra committed:
change collector pod name to eraser

ashnamehrotra committed:
change eraser container and image name to remover

ashnamehrotra committed:
change eraser container references to remover

ashnamehrotra committed:
update references in Makefile and Dockerfile

ashnamehrotra committed:
update Dockerfile.manifest

ashnamehrotra committed:
make image name for tests more specific

ashnamehrotra committed:
update references in e2e tests

ashnamehrotra committed:
update eraser metrics to remover metrics

ashnamehrotra committed:
update test.yaml with remover

ashnamehrotra committed:
make IMAGE specific to REMOVER_IMAGE

ashnamehrotra committed:
update controller manager config

ashnamehrotra committed:
update manifest_staging

ashnamehrotra committed:
update API

ashnamehrotra committed:
update func name metrics test

ashnamehrotra committed:
Merge branch 'main' of https://github.com/Azure/eraser into rename-collector-pods

ashnamehrotra committed:
change to remover in test.yaml

ashnamehrotra committed:
fix workflows

ashnamehrotra committed:
change util.Image to util.RemoverImage

ashnamehrotra committed:
fix e2e test errors

ashnamehrotra committed:
update manifest

ashnamehrotra committed:
fix metrics test scanner

ashnamehrotra committed:
Merge branch 'main' of https://github.com/Azure/eraser into rename-collector-pods

ashnamehrotra committed:
Merge branch 'main' of https://github.com/Azure/eraser into rename-collector-pods

ashnamehrotra committed:
Merge branch 'main' of https://github.com/Azure/eraser into rename-collector-pods

pmengelbert commented on api/v1alpha1/config/config.go:
We'll also need this for v1, right?

sozercan replied:
is this user facing or backwards compatible?

pmengelbert replied:
this is user-facing, since this defines the configmap keys

pmengelbert replied:
> We'll also need this for v1, right?

It looks like this file doesn't exist in `api/v1`, so obviously this can't be changed there!

sozercan replied:
why isn't this in v1?

sozercan replied:
ah nm, this is the alpha configmap change. that makes sense

pmengelbert commented:
Otherwise LGTM

pmengelbert commented:
Actually, I noticed that `test/e2e/helm-empty-values.yaml` still says `eraser`

pmengelbert commented:
Should we also fix the docs?

ashnamehrotra committed:
fix helm test eraser reference

ashnamehrotra committed:
update docs

ashnamehrotra committed:
update values file

ashnamehrotra committed:
fix upgrade workflow

ashnamehrotra commented:
helm install with repo fails, converting to draft. Helm test is incorrectly passing since eraser image is not set.

ashnamehrotra committed:
add v1alpha2 api

ashnamehrotra committed:
update manifest_staging

ashnamehrotra committed:
Merge branch 'main' of https://github.com/Azure/eraser into rename-collector-pods

ashnamehrotra committed:
update

ashnamehrotra committed:
Merge branch 'main' of https://github.com/Azure/eraser into rename-collector-pods

ashnamehrotra committed:
fix workflow files

ashnamehrotra committed:
fix install helm latest release test to use remover image

ashnamehrotra committed:
fix typo

ashnamehrotra committed:
increase e2e-test timeout

ashnamehrotra committed:
cleanup

ashnamehrotra committed:
reset e2e test timeout

pmengelbert commented on test/e2e/util/utils.go:
in `HelmDeployLatestEraserRelease`, we will want to install helm using `-f <helm-empy-values>` to be sure we are not using the defaults. The `-f` should come before all of the `--set` arguments.
```
		emptyValuesPath, err := filepath.Abs(filepath.Join(wd, "../../test-data/helm-empty-values.yaml"))
		if err != nil {
			return ctx, err
		}

		// start deployment
		allArgs := []string{providerResourceAbsolutePath, "-f", emptyValuesPath}
```

pmengelbert commented on api/v1alpha1/eraserconfig_types.go:
```diff
@@ -135,7 +135,7 @@ type RepoTag struct {
 type Components struct {
 	Collector OptionalContainerConfig `json:"collector,omitempty"`
 	Scanner   OptionalContainerConfig `json:"scanner,omitempty"`
-	Eraser    ContainerConfig         `json:"eraser,omitempty"`
```
We'll need to revert this back to `Eraser` and `eraser`

pmengelbert commented on api/v1alpha1/config/config.go:
```diff
@@ -136,9 +136,9 @@ func Default() *v1alpha1.EraserConfig {
 					Config: &defaultScannerConfig,
 				},
 			},
-			Eraser: v1alpha1.ContainerConfig{
```
Ditto here (revert to eraser instead of remover)

pmengelbert commented on test/e2e/tests/collector_skip_excluded/main_test.go:
```diff
@@ -40,13 +40,15 @@ func TestMain(m *testing.M) {
 		util.HelmDeployLatestEraserRelease(util.TestNamespace,
 			"--set", util.ScannerEnable.Set("false"),
 			"--set", util.CollectorEnable.Set("false"),
+			"--set", util.RemoverImageRepo.Set(removerImage.Repo),
+			"--set", util.RemoverImageTag.Set(removerImage.Tag),
```
Set the eraser image instead of the remover image, since the old chart uses v1alpha1.

pmengelbert reviewed (changes requested):
Change v1alpha1 to v1alpha2 in most (all?) places, so that the configmap changes will take effect.

pmengelbert commented on config/manager/controller_manager_config.yaml:
We'll need to change `apiVersion: eraser.sh/v1alpha1` to `v1alpha2` here

ashnamehrotra committed:
update versions to v1alpha2

ashnamehrotra committed:
revert v1alpha1 api to original

ashnamehrotra committed:
add config to v1alpha2

ashnamehrotra committed:
use emptyValues file in HelmDeployLatestEraserRelease

ashnamehrotra committed:
update tests to accomodate different versions

ashnamehrotra committed:
fixes

ashnamehrotra committed:
change more v1alpha1 references

ashnamehrotra committed:
change api in main

ashnamehrotra committed:
set manager image in HelmDeployLatestEraserRelease

ashnamehrotra committed:
increase timeouts

ashnamehrotra committed:
changes for using eraser image for tests for v1alpha1

ashnamehrotra committed:
lint fix

ashnamehrotra committed:
fix workflows to incorporate eraser image again

ashnamehrotra committed:
attempt at unversioned config

ashnamehrotra committed:
remove scheme builder

ashnamehrotra committed:
fix log

sozercan commented on controllers/imagecollector/imagecollector_controller.go:
```diff
@@ -375,7 +375,7 @@ func (r *Reconciler) createImageJob(ctx context.Context) (ctrl.Result, error) {
 	}
 
-	job := &eraserv1alpha1.ImageJob{
+	job := &eraserv1alpha2.ImageJob{
```
i think this was intentionally left on v1alpha1?

ashnamehrotra committed:
revert ImageJob change to v1alpha2

pmengelbert committed:
Initial conversion-gen work

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Initial conversion-gen work

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Add v1 to conversions

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Add v1 to conversions

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Allow defaults for v1alpha2/config

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Allow defaults for v1alpha2/config

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Add defaults for api/unversioned/config

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Add defaults for api/unversioned/config

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Use unversioned in main.go and add groupversion_info.go to unversioned and v1alpha2

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Use unversioned in main.go and add groupversion_info.go to unversioned and v1alpha2

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Update main.go to correctly unmarshal config

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Update main.go to correctly unmarshal config

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Fix build/version.sh to accept BuildVersion

The build/version.sh script was setting `buildVersion` instead of
`BuildVersion`. This was causing the default values for tags to be empty
strings, which is what we want, but not how we want to get there.

The tags will be provided by either the helm chart or the manifest.
Thus, in order for them to be overwritten during unmarshaling, they must
be empty in the default config.

This commit explicitly makes the image tags empty.

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Fix build/version.sh to accept BuildVersion

The build/version.sh script was setting `buildVersion` instead of
`BuildVersion`. This was causing the default values for tags to be empty
strings, which is what we want, but not how we want to get there.

The tags will be provided by either the helm chart or the manifest.
Thus, in order for them to be overwritten during unmarshaling, they must
be empty in the default config.

This commit explicitly makes the image tags empty.

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Supply defaults properly

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Supply defaults properly

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Collapse the defaults correctly

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Collapse the defaults correctly

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Go mod tidy

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Go mod tidy

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Fix linter errors

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Fix linter errors

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Remove unused import

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Remove unused import

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Return logging to debug

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Return logging to debug

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Create convertFunc type

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Create convertFunc type

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Fix mental error in thinking about defaults

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Fix mental error in thinking about defaults

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Remove unnecessary yaml tags

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Remove unnecessary yaml tags

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Fix gnarly commands in Make

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Fix gnarly commands in Make

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

ashnamehrotra committed:
Merge branch 'fix/conversion-gen' of https://github.com/pmengelbert/eraser into rename-collector-pods

pmengelbert committed:
Address PR comments

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Combine `replace` in go.mod

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

ashnamehrotra committed:
Merge branch 'fix/conversion-gen' of https://github.com/pmengelbert/eraser into rename-collector-pods

ashnamehrotra committed:
missing from merge conflict

ashnamehrotra committed:
make generate

ashnamehrotra committed:
lint fix

ashnamehrotra committed:
change unversioned api repo image name

ashnamehrotra committed:
change apiVersion for helm charts

ashnamehrotra committed:
add apiVersion to helm-empty-values

pmengelbert committed:
Default eraser -> remover

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

ashnamehrotra committed:
test

ashnamehrotra committed:
Merge branch 'fix/conversion-gen' of https://github.com/pmengelbert/eraser into rename-collector-pods

ashnamehrotra committed:
fix import

ashnamehrotra committed:
save

ashnamehrotra committed:
delete unused

ashnamehrotra commented:
It looks like the tests were failing because the collector controller was never added to the manager since when HelmDeployLatesetEraserRelease() was called in the tests it would set collector enabled to "false", and in lines [95-99](rename-collector-pods) we don't initially add the collector if that is the case. So any configmap update that later wants to use the collector will not work. The tests are passing locally now, but they aren't working in the CI which could be due to the delay since now it will deploy the pods for HelmDeployLatestEraserRelease() before UpgradeEraserHelm()

pmengelbert commented:
> It looks like the tests were failing because the collector controller was never added to the manager since when HelmDeployLatesetEraserRelease() was called in the tests it would set collector enabled to "false", and in lines [95-99](rename-collector-pods) we don't initially add the collector if that is the case. So any configmap update that later wants to use the collector will not work. The tests are passing locally now, but they aren't working in the CI which could be due to the delay since now it will deploy the pods for HelmDeployLatestEraserRelease() before UpgradeEraserHelm()

What files are those line numbers for?

I believe disabling the collector and scanner is correct, because otherwise a job will start running immediately. The configmap update won't happen if a job is running, so none of the updates will take effect. In that case, we are testing the older version of eraser (the one in the repo), and not the latest version.

ashnamehrotra commented:
> 

It is for imagecollector-controller. It never ends up adding the imagecollector-controller to the manager because it reaches this line and returns nil:

```
	collCfg := c.Components.Collector
	if !collCfg.Enabled {
		// don't add controller, but don't throw an error either
		return nil
	}

```

Even though the configmap updates correctly, it doesn't reach the reconcile.

pmengelbert commented:
> > 
> 
> It is for imagecollector-controller. It never ends up adding the imagecollector-controller to the manager because it reaches this line and returns nil:
> 
> ```
> 	collCfg := c.Components.Collector
> 	if !collCfg.Enabled {
> 		// don't add controller, but don't throw an error either
> 		return nil
> 	}
> ```
> 
> Even though the configmap updates correctly, it doesn't reach the reconcile.

I see. Great work investigating this.

I think what we need to do is if the collector or scanner are enabled after being initially disabled, we need to immediately `os.Exit(1)` (with a log). This will crash the manager, but it will restart immediately and initialize with the new config.

This would need to be done in `startConfigWatch`. Something like this:
```diff
diff --git a/main.go b/main.go
index b6b6de81..10befa50 100644
--- a/main.go
+++ b/main.go
@@ -248,13 +248,19 @@ func startConfigWatch(watcher *inotify.Watcher, eraserOpts *config.Manager, file
 				continue
 			}
 
-			cfg, err := getConfig(filename)
+			oldCfg, err := eraserOpts.Read()
+			if err != nil {
+				setupLog.Error(err, "configuration could not be read", "event", ev, "filename", filename)
+
+			}
+
+			newCfg, err := getConfig(filename)
 			if err != nil {
 				setupLog.Error(err, "configuration is missing or invalid", "event", ev, "filename", filename)
 				continue
 			}
 
-			if err = eraserOpts.Update(cfg); err != nil {
+			if err = eraserOpts.Update(newCfg); err != nil {
 				setupLog.Error(err, "configuration update failed")
 				continue
 			}
@@ -265,6 +271,18 @@ func startConfigWatch(watcher *inotify.Watcher, eraserOpts *config.Manager, file
 				continue
 			}
 
+			type check struct {
+				c bool
+				s bool
+			}
+
+			old := check{c: oldCfg.Components.Collector.Enabled, s: oldCfg.Components.Scanner.Enabled}
+			nnew := check{c: newC.Components.Collector.Enabled, s: newC.Components.Scanner.Enabled}
+			if old != nnew {
+				setupLog.Info("configurations differ in an irreconcileable way", "old", old, "new", nnew)
+				os.Exit(1)
+			}
+
 			setupLog.V(1).Info("new configuration", "manager", newC.Manager, "components", newC.Components)
 		case err := <-watcher.Error:
 			setupLog.Error(err, "file watcher error")
```

@sozercan thoughts on this approach?

ashnamehrotra committed:
implement check for different configs

ashnamehrotra committed:
Merge branch 'main' of https://github.com/Azure/eraser into rename-collector-pods

ashnamehrotra committed:
integrate context for exit

ashnamehrotra committed:
Merge branch 'main' of https://github.com/Azure/eraser into rename-collector-pods

ashnamehrotra committed:
increase utils.Timeout

ashnamehrotra committed:
Merge branch 'main' of https://github.com/Azure/eraser into rename-collector-pods

pmengelbert commented on main.go:
```diff
@@ -265,6 +280,18 @@ func startConfigWatch(watcher *inotify.Watcher, eraserOpts *config.Manager, file
+			}
+
+			old := check{collector: oldCfg.Components.Collector.Enabled, sscanner: oldCfg.Components.Scanner.Enabled}
+			new := check{collector: newC.Components.Collector.Enabled, sscanner: newC.Components.Scanner.Enabled}
```
`new` is a keyword in golang, so you should use another variable name (this is failing the linter)

pmengelbert commented on .github/workflows/e2e-build.yaml:
```diff
@@ -8,6 +8,46 @@ on:
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ inputs.bucket-id }}
+          path: remover_test.tar     
```
This should replace `build-eraser`, but you've left both.

ashnamehrotra replied:
we need to build both eraser and remover for the e2e tests (same as below)

pmengelbert commented on .github/workflows/e2e-test.yaml:
```diff
@@ -73,29 +73,34 @@ jobs:
-          echo ERASER_TAG=$ERASER_TAG >> $GITHUB_ENV
+          echo REMOVER_TAG=$REMOVER_TAG >> $GITHUB_ENV
           echo MANAGER_TAG=$MANAGER_TAG >> $GITHUB_ENV
+          echo ERASER_TAG=$ERASER_TAG >> $GITHUB_ENV
```
Why have both `ERASER` and `REMOVER`?

ashnamehrotra replied:
for the e2e tests, HelmDeployLatestEraserRelease() uses Eraser before we call UpgradeEraserHelm() with Remover

ashnamehrotra committed:
rename new var

ashnamehrotra committed:
run linter

sozercan commented on charts/eraser/values.yaml:
```diff
@@ -69,7 +69,7 @@ runtimeConfig:
         # timeout:
         #   total: 23h
         #   perImage: 1h
-    eraser:
```
this shouldn't need to be updated. it will be updated automatically when we do a release

ashnamehrotra committed:
revert charts/eraser/values.yaml

ashnamehrotra committed:
remove docker-build-eraser

ashnamehrotra committed:
remove docker-build-eraser from makefile

pmengelbert committed:
Save before sed

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
lots of sed

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Remove build eraser stage from test workflow

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Rename EraserImage{Repo,Tag} to RemoverImage{Repo,Tag}

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
A little refactoring

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Remove cache for e2e-test only, since we are not building multiple times

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Remove unused actions from e2e-test.yaml

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Restore e2e timeout to 5min

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Save before sed

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Remove GetManagerLogs calls, since we are fetching ALL logs from the cluster in e2e tests

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Save before sed

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Simplify signature of GetPodLogs

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Update test name to avoid conflict

Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

pmengelbert committed:
Merge remote-tracking branch 'upstream/main' into peter/test-rename

pmengelbert commented on api/v1alpha2/imagejob_types.go:
This file should be deleted and anything referencing the types in this file should use `v1` instead

pmengelbert commented on api/v1alpha2/imagelist_types.go:
This file should be deleted. Anything referencing the types in this file should instead reference `v1`

pmengelbert commented on controllers/imagecollector/imagecollector_controller.go:
```diff
@@ -455,7 +456,7 @@ func (r *Reconciler) createImageJob(ctx context.Context) (ctrl.Result, error) {
 			Namespace: namespace,
 			OwnerReferences: []metav1.OwnerReference{
-				*metav1.NewControllerRef(job, eraserv1alpha1.GroupVersion.WithKind("ImageJob")),
+				*metav1.NewControllerRef(job, eraserv1alpha2.GroupVersion.WithKind("ImageJob")),
```
Use `v1`

pmengelbert commented on controllers/imagejob/imagejob_controller.go:
```diff
@@ -117,7 +117,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		},
 		&handler.EnqueueRequestForOwner{
-			OwnerType:    &eraserv1alpha1.ImageJob{},
+			OwnerType:    &eraserv1alpha2.ImageJob{},
```
`v1`

pmengelbert commented on controllers/imagejob/imagejob_controller.go:
```diff
@@ -130,7 +130,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		},
 		&handler.EnqueueRequestForOwner{
-			OwnerType:    &eraserv1alpha1.ImageJob{},
+			OwnerType:    &eraserv1alpha2.ImageJob{},
```
`v1`

pmengelbert commented on controllers/imagelist/imagelist_controller.go:
```diff
@@ -331,7 +331,7 @@ func (r *Reconciler) handleImageListEvent(ctx context.Context, imageList *eraser
 			},
 			OwnerReferences: []metav1.OwnerReference{
-				*metav1.NewControllerRef(imageList, eraserv1alpha1.GroupVersion.WithKind("ImageList")),
+				*metav1.NewControllerRef(imageList, eraserv1alpha2.GroupVersion.WithKind("ImageList")),
```
`v1`

pmengelbert commented on controllers/imagelist/imagelist_controller.go:
```diff
@@ -370,7 +370,7 @@ func (r *Reconciler) handleImageListEvent(ctx context.Context, imageList *eraser
 			Namespace: utils.GetNamespace(),
 			OwnerReferences: []metav1.OwnerReference{
-				*metav1.NewControllerRef(job, eraserv1alpha1.GroupVersion.WithKind("ImageJob")),
+				*metav1.NewControllerRef(job, eraserv1alpha2.GroupVersion.WithKind("ImageJob")),
```
`v1`

pmengelbert commented on controllers/imagelist/imagelist_controller.go:
```diff
@@ -381,7 +381,7 @@ func (r *Reconciler) handleImageListEvent(ctx context.Context, imageList *eraser
 	}
 
-	configMap.ObjectMeta.OwnerReferences = []metav1.OwnerReference{*metav1.NewControllerRef(job, eraserv1alpha1.GroupVersion.WithKind("ImageJob"))}
+	configMap.ObjectMeta.OwnerReferences = []metav1.OwnerReference{*metav1.NewControllerRef(job, eraserv1alpha2.GroupVersion.WithKind("ImageJob"))}
```
`v1`

pmengelbert commented on pkg/scanners/trivy/helpers.go:
```diff
@@ -27,7 +27,7 @@ func loadConfig(filename string) (Config, error) {
 	}
 
-	var eraserConfig eraserv1alpha1.EraserConfig
+	var eraserConfig eraserv1alpha2.EraserConfig
```
This should be `unversioned`

pmengelbert commented on pkg/scanners/trivy/types.go:
```diff
@@ -41,8 +41,8 @@ type (
-		Total    eraserv1alpha1.Duration `json:"total,omitempty"`
-		PerImage eraserv1alpha1.Duration `json:"perImage,omitempty"`
+		Total    eraserv1alpha2.Duration `json:"total,omitempty"`
+		PerImage eraserv1alpha2.Duration `json:"perImage,omitempty"`
```
both `unversioned`

pmengelbert commented on pkg/scanners/trivy/types.go:
```diff
@@ -81,8 +81,8 @@ func DefaultConfig() *Config {
-			Total:    eraserv1alpha1.Duration(time.Hour * 23),
-			PerImage: eraserv1alpha1.Duration(time.Hour),
+			Total:    eraserv1alpha2.Duration(time.Hour * 23),
+			PerImage: eraserv1alpha2.Duration(time.Hour),
```
both `unversioned`

pmengelbert commented on test/e2e/tests/helm_pull_secret_imagelist/eraser_test.go:
```diff
@@ -20,22 +20,22 @@ import (
 			}
 
-			imgList := &eraserv1alpha1.ImageList{
+			imgList := &eraserv1alpha2.ImageList{
```
`v1`

pmengelbert commented on test/e2e/tests/helm_pull_secret_imagelist/eraser_test.go:
```diff
@@ -20,22 +20,22 @@ import (
+			imgList := &eraserv1alpha2.ImageList{
 				ObjectMeta: metav1.ObjectMeta{Name: util.Prune},
-				Spec: eraserv1alpha1.ImageListSpec{
+				Spec: eraserv1alpha2.ImageListSpec{
```
`v1`

pmengelbert commented on test/e2e/tests/imagelist_alias/eraser_test.go:
```diff
@@ -118,9 +118,9 @@ func TestEnsureAliasedImageRemoved(t *testing.T) {
 		}).
 		Assess("Image deleted when referencing by alias", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
-			imgList := &eraserv1alpha1.ImageList{
+			imgList := &eraserv1alpha2.ImageList{
```
`v1`

pmengelbert commented on test/e2e/tests/imagelist_alias/eraser_test.go:
```diff
@@ -118,9 +118,9 @@ func TestEnsureAliasedImageRemoved(t *testing.T) {
+			imgList := &eraserv1alpha2.ImageList{
 				ObjectMeta: metav1.ObjectMeta{Name: util.Prune},
-				Spec: eraserv1alpha1.ImageListSpec{
+				Spec: eraserv1alpha2.ImageListSpec{
```
`v1`

pmengelbert commented on test/e2e/tests/imagelist_change/eraser_test.go:
```diff
@@ -115,7 +115,7 @@ func TestUpdateImageList(t *testing.T) {
 		Assess("Deploy imagelist to remove nginx", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
 			// deploy imageJob config
-			if err := util.DeployEraserConfig(cfg.KubeconfigFile(), cfg.Namespace(), "../../test-data", "eraser_v1alpha1_imagelist.yaml"); err != nil {
+			if err := util.DeployEraserConfig(cfg.KubeconfigFile(), cfg.Namespace(), "../../test-data", "eraser_v1alpha2_imagelist.yaml"); err != nil {
```
`"eraser_v1_imagelist.yaml"`

pmengelbert commented on test/e2e/tests/imagelist_change/eraser_test.go:
```diff
@@ -127,7 +127,7 @@ func TestUpdateImageList(t *testing.T) {
 		Assess("Update imagelist to prune rest of images", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
 			// deploy imageJob config
-			if err := util.DeployEraserConfig(cfg.KubeconfigFile(), cfg.Namespace(), "../../test-data", "eraser_v1alpha1_imagelist_updated.yaml"); err != nil {
+			if err := util.DeployEraserConfig(cfg.KubeconfigFile(), cfg.Namespace(), "../../test-data", "eraser_v1alpha2_imagelist_updated.yaml"); err != nil {
```
`"eraser_v1_imagelist_updated.yaml"`, there should be no imagelist yaml file with `v1alpha2` in the title, as the `v1alpha2/imagelist_types` should be removed (we already have v1).

pmengelbert commented on test/e2e/tests/imagelist_include_nodes/eraser_test.go:
```diff
@@ -115,12 +115,12 @@ func TestIncludeNodes(t *testing.T) {
 
 			// deploy imageJob config
-			if err = util.DeployEraserConfig(cfg.KubeconfigFile(), cfg.Namespace(), "../../test-data", "eraser_v1alpha1_imagelist.yaml"); err != nil {
+			if err = util.DeployEraserConfig(cfg.KubeconfigFile(), cfg.Namespace(), "../../test-data", "eraser_v1alpha2_imagelist.yaml"); err != nil {
```
`"eraser_v1_imagelist.yaml"`

ashnamehrotra committed:
change versioning - remove v1alpha2 for ImageList and ImageJob

ashnamehrotra committed:
set v1 api as storage version for imagejob and imagelist

pmengelbert commented on test/e2e/tests/imagelist_change/eraser_test.go:
```diff
@@ -127,7 +127,7 @@ func TestUpdateImageList(t *testing.T) {
 		Assess("Update imagelist to prune rest of images", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
 			// deploy imageJob config
-			if err := util.DeployEraserConfig(cfg.KubeconfigFile(), cfg.Namespace(), "../../test-data", "eraser_v1alpha2_imagelist_updated.yaml"); err != nil {
+			if err := util.DeployEraserConfig(cfg.KubeconfigFile(), cfg.Namespace(), "../../test-data", "eraser_v1_imagelist_updated.yaml"); err != nil {
```
This file needs to exist first

ashnamehrotra committed:
rename eraser_v1alpha2_imagelist_updated to v1, delete eraser_v1alpha2_imagelist

ashnamehrotra committed:
update manifest

ashnamehrotra committed:
add v1 to scheme in failing tests

sozercan commented on .github/workflows/e2e-build.yaml:
```diff
@@ -8,46 +8,46 @@ on:
           egress-policy: audit
-      - name: Set up Go 1.20
-        uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9 # v4.0.0
+      - name: Set up Go 1.19
```
looks like these are getting downgraded

sozercan commented on api/v1alpha1/eraserconfig_types.go:
```diff
@@ -76,7 +76,7 @@ func (r *Runtime) UnmarshalJSON(b []byte) error {
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+// NOTE: yaml tags are required.  Any new fields you add must have yaml tags for the fields to be serialized.
```
i thought we reverted this change

pmengelbert replied:
We did, so this should be reverted here too.

sozercan commented on manifest_staging/charts/eraser/templates/imagejobs.eraser.sh-customresourcedefinition.yaml:
```diff
@@ -115,3 +115,50 @@ spec:
     storage: false
     subresources:
       status: {}
+  - name: v1alpha2
```
why do we have v1alpha2 of imagejob and imagelists?

pmengelbert replied:
We shouldn't

pmengelbert replied:
I think re-running codegen and manifest gen should fix this, since those files were removed.

sozercan commented on docs/docs/customization.md:
```diff
@@ -72,12 +72,12 @@ take place when:
 ### Swapping out components
 
-The collector, scanner, and eraser components can all be swapped out. This
+The collector, scanner, and remover components can all be swapped out. This
```
i think we need to change these in versioned docs?

pmengelbert replied:
Good catch

ashnamehrotra replied:
should I change in both docs/ and version-v1.0.x/ or just version-v1.0.x/?

sozercan replied:
oh my mistake, `docs` becomes the vNext docs

pmengelbert commented on test/e2e/test-data/eraser_v1_imagelist_updated.yaml:
```diff
@@ -1,4 +1,4 @@
-apiVersion: eraser.sh/v1alpha1
+apiVersion: eraser.sh/v1alpha2
```
Should be `v1`

sozercan replied:
i think we kept this in `v1alpha1` intentionally to test conversion

ashnamehrotra replied:
would it be better to revert this back to eraser_v1alpha1_imagelist_updated?

pmengelbert replied:
This hasn't been resolved yet

ashnamehrotra committed:
fix downgrade

ashnamehrotra committed:
revert yaml to json in v1alpha1

ashnamehrotra committed:
rerun codegen and manifest gen

sozercan commented on .github/workflows/e2e-build.yaml:
```diff
@@ -8,46 +8,46 @@ on:
+        run: 'make docker-build-remover OUTPUT_TYPE=type=oci,dest=./${REMOVER_REPO}_${REMOVER_TAG}.tar,name=${REMOVER_REPO}:${REMOVER_TAG}'
       - name: Upload Build Artifacts
-        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+        uses: actions/upload-artifact@v3
```
let's keep the digest here

sozercan commented on .github/workflows/e2e-build.yaml:
```diff
@@ -8,46 +8,46 @@ on:
+      - uses: crazy-max/ghaction-github-runtime@v2
       - name: Check out code into the Go module directory
-        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+        uses: actions/checkout@v3
```
same here

ashnamehrotra committed:
keep digest

ashnamehrotra committed:
fix typo

ashnamehrotra committed:
rename updated imagelist for test back to v1alpha2

ashnamehrotra committed:
imagelist_alpine to v1

ashnamehrotra committed:
change back to eraser_v1alpha1_imagelist_updated.yaml

sozercan commented on .github/workflows/e2e-build.yaml:
```diff
@@ -8,46 +8,46 @@ on:
           go-version: "1.20"
       - name: Setup buildx instance
-        uses: docker/setup-buildx-action@4b4e9c3e2d4531116a6f8ba8e71fc6e2cb6e6c8c # v2.5.0
+        uses: docker/setup-buildx-action@v2
```
let's revert to use the hash

sozercan commented on .github/workflows/e2e-build.yaml:
```diff
@@ -8,46 +8,46 @@ on:
         with:
           use: true
-      - uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
+      - uses: actions/cache@v3.3.1
```
let's revert to use the hash

sozercan commented on .github/workflows/e2e-build.yaml:
```diff
@@ -8,46 +8,46 @@ on:
             ~/go/pkg/mod
             ~/.cache/go-build
-      - uses: crazy-max/ghaction-github-runtime@747e98cd97bb9cfed8fe0165c821a83139bfdd67 # v2.1.0
+      - uses: crazy-max/ghaction-github-runtime@v2
```
let's revert to use the hash

ashnamehrotra committed:
revert to use hash

ashnamehrotra committed:
revert harde runner hash

ashnamehrotra committed:
get latest from v1alpha1 config

ashnamehrotra committed:
remove whitespace

ashnamehrotra committed:
Merge branch 'main' of https://github.com/Azure/eraser into rename-collector-pods

ashnamehrotra committed:
revert test-data files besides helm-empty-values

ashnamehrotra committed:
revert test file names back to original

pmengelbert commented on main.go:
```diff
@@ -248,26 +257,51 @@ func startConfigWatch(watcher *inotify.Watcher, eraserOpts *config.Manager, file
-			setupLog.V(1).Info("new configuration", "manager", newC.Manager, "components", newC.Components)
+			if needsRestart(oldConfig, newConfig) {
+				setupLog.Info("configurations differ in an irreconcileable way, restarting", "old", oldConfig.Components, "new", newConfig.Components)
+				cancel()
```
```suggestion
                                // restarts the manager
				cancel()
```

pmengelbert commented on pkg/scanners/trivy/helpers.go:
```diff
@@ -27,7 +27,7 @@ func loadConfig(filename string) (Config, error) {
 	}
 
-	var eraserConfig eraserv1alpha1.EraserConfig
+	var eraserConfig unversioned.EraserConfig
```
This should be addressed in a separate PR, but we will want to manage a versioned schema for the scanner config as well.

ashnamehrotra committed:
add restarts manager comment

pmengelbert commented on test/e2e/tests/collector_disable_scan/main_test.go:
```diff
@@ -17,9 +17,9 @@ import (
 
 func TestMain(m *testing.M) {
-	utilruntime.Must(eraserv1alpha1.AddToScheme(scheme.Scheme))
+	utilruntime.Must(eraserv1alpha2.AddToScheme(scheme.Scheme))
```
I don't know what this line is doing. I'm not sure if it should be `v1alpha1` or `v1alpha2`.

ashnamehrotra committed:
test with revert change from v1alpha2 add to scheme

ashnamehrotra committed:
revert cont

ashnamehrotra committed:
remove eraserv1alpha2.AddToScheme from main.go

ashnamehrotra committed:
register v1 for eraserv1.ImageList tests

ashnamehrotra committed:
typo fix
